src/helpers/Constants.sol

```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.25;
+
+// Index marking the start of the data for the validation function.
+uint8 constant RESERVED_VALIDATION_DATA_INDEX = 255;
+
+// Magic value for the Entity ID of direct call validation.
+uint32 constant SELF_PERMIT_VALIDATION_FUNCTIONID = type(uint32).max;
```

Comment:
```suggestion
uint32 constant SELF_PERMIT_VALIDATION_ENTITYID = type(uint32).max;
```

Reply:
Good catch. Should we rename this even further, to `DIRECT_CALL_VALIDATION_ENTITYID`?

Reply:
Maybe even add module to it. `MODULE_DIRECT_CALL_VALIDATION_ENTITYID`.

Reply:
Technically it doesn't have to be a module, you could add an EOA, but it's a less likely use case